### DISCUSSION TITLE
Does Ugrep support Searching across files

### DISCUSSION THREAD

**Discussion by sharrythakur** [Genivia/ugrep#422] [Q&A] [answered]
I have a pattern **'Sushant AND Thakur'**

I want to run ugrep on a directory and i want the output if **Sushant** is present in one file and **Thakur** is present in other files under the directory.
Basically return the results only if the sub-patterns are found in different files

**Comment by genivia-inc** [comment, marked as answer]
This can be done with a bash script to search for one pattern in a file in a directory and when found in a directory then search for the other pattern in that directory. Grep tools exit with code 0 when a file matches so it can be used in a `if-then` to invoke the second grep.
